pkg/helm/client.go

```diff
@@ -258,296 +208,439 @@ func (h *HelmClient) PullByRefWithRetries(ctx context.Context, ref string, versi
-		return nil, fmt.Errorf("run render: %w", err)
+	// Add kubernetes version if available
+	if h.kversion != nil {
+		args = append(args, "--kube-version", fmt.Sprintf("v%d.%d.0", h.kversion.Major(), h.kversion.Minor()))
 	}
 
-	var manifests bytes.Buffer
-	fmt.Fprintln(&manifests, strings.TrimSpace(release.Manifest))
-	for _, m := range release.Hooks {
-		fmt.Fprintf(&manifests, "---\n# Source: %s\n%s\n", m.Path, m.Manifest)
-	}
+	// Add kubeconfig and context if available
+	args = h.addKubernetesEnvArgs(args)
 
-	resources := [][]byte{}
-	splitManifests := releaseutil.SplitManifests(manifests.String())
-	for _, manifest := range splitManifests {
-		manifest = strings.TrimSpace(manifest)
-		resources = append(resources, []byte(manifest))
-	}
+	// Add include CRDs flag
+	args = append(args, "--include-crds")
 
-	return resources, nil
-}
+	// Add debug flag to report progress and capture debug logs
+	args = append(args, "--debug")
 
-func (h *HelmClient) getActionCfg(namespace string) (*action.Configuration, error) {
-	cfg := &action.Configuration{}
-	var logFn action.DebugLog
-	if h.logFn != nil {
-		logFn = h.logFn
-	} else {
-		logFn = _logFn
-	}
-	restClientGetter := &namespacedRESTClientGetter{
-		RESTClientGetter: h.restClientGetter,
-		namespace:        namespace,
+	// Execute helm template command
+	stdout, stderr, err := h.executor.ExecuteCommand(ctx, nil, args...)
+	if err != nil {
+		return nil, fmt.Errorf("helm template: %w, stderr: %s", err, stderr)
 	}
-	if err := cfg.Init(restClientGetter, namespace, "secret", logFn); err != nil {
-		return nil, fmt.Errorf("init helm configuration: %w", err)
+
+	manifests, err := splitManifests(stdout)
+	if err != nil {
+		return nil, fmt.Errorf("parse helm template output: %w", err)
 	}
-	return cfg, nil
+	return manifests, nil
 }
 
-func (h *HelmClient) loadChart(ctx context.Context, releaseName, chartPath, chartVersion string) (*chart.Chart, error) {
-	var localPath string
-	if h.airgapPath != "" {
-		// airgapped, use chart from airgap path
-		// TODO: this should just respect the chart path if it's a local path and leave it up to the caller to handle
-		localPath = filepath.Join(h.airgapPath, fmt.Sprintf("%s-%s.tgz", releaseName, chartVersion))
-	} else if !strings.HasPrefix(chartPath, "/") {
-		// Assume this is a chart from a repo if it doesn't start with a /
-		// This includes oci:// prefix
-		var err error
-		localPath, err = h.PullByRefWithRetries(ctx, chartPath, chartVersion, 3)
-		if err != nil {
-			return nil, fmt.Errorf("pull: %w", err)
-		}
-		defer os.RemoveAll(localPath)
-	} else {
-		localPath = chartPath
+// addKubernetesEnvArgs adds kubernetes environment arguments to the helm command
+func (h *HelmClient) addKubernetesEnvArgs(args []string) []string {
+	if h.kubernetesEnvSettings == nil {
+		return args
 	}
 
-	chartRequested, err := loader.Load(localPath)
-	if err != nil {
-		return nil, fmt.Errorf("load: %w", err)
+	// Add all helm CLI flags from kubernetesEnvSettings
```

Comment:
could you move addKubernetesCLIFlags to this package and export it or make this dynamic?

Reply:
can you elaborate? what do you mean by making it dynamic?